ops/prod/metabase.tf

```diff
@@ -14,4 +14,30 @@ module "metabase" {
 
   postgres_server_name = data.terraform_remote_state.persistent_prod.outputs.postgres_server_name
   postgres_url         = "@Microsoft.KeyVault(SecretUri=${data.azurerm_key_vault_secret.metabase_db_uri.id})"
+}
+
+module "metabase-readonly" {
+  source = "../services/app_service/metabase"
+  name   = "${local.project}-${local.name}-${local.env}-metabase-readonly"
+  env    = local.env
+
+  app_settings = {
+    "MB_DB_USER"           = data.azurerm_key_vault_secret.postgres_nophi_user.value
```

Comment:
We'll probably need to talk through this one, but we have a circular dependency if I'm reading this right:
* This PR is adding `ops/services/app_service/metabase/vault.tf` file to handle creating the `postgres_nophi_user` and `postgres_nophi_pass` resources.
* But to create this module, we need the output from those resources, but that output only exists if the module is run, and so on.

Reply:
Hmmm yeah, I see what you mean. I think the circular dependency was probably introduced when I changed the `MB_DB_USER` from being a "plain" Azure secret to being created in the `vault.tf` file. Do you think it would work to break out the `nophi_user/pass` vault creation into another module? It doesn't have any dependencies beyond the env and global_vault_id, so if it was in another module and that module was executed first, maybe that would work.

Reply:
Yeah I think there are two options here (maybe 3?):
* Make these secrets part of another module, like you suggested
* Have the module export these values as `output`s, and read them that way rather than a data source
  * Pros: clean and makes logical sense
  * Cons: we would expose these variables in the remote state file - if we're encrypting it and correctly restricting access is that an actual problem?
* Bonus potential third approach: can we use a service principal like the main Metabase is doing? I legit don't know, but I'm hoping to dig into this later today.